sdk/storage/storage-file-share/src/ShareClient.ts

```diff
@@ -988,12 +988,22 @@ export class ShareClient extends StorageClient {
       };
 
       for (const identifier of response) {
+        let accessPolicy: any = undefined;
+        if (identifier.accessPolicy) {
+          accessPolicy = {
+            permissions: identifier.accessPolicy.permissions
+          };
+
+          if (identifier.accessPolicy.expiresOn) {
+            accessPolicy.expiresOn = new Date(identifier.accessPolicy.expiresOn);
+          }
+
+          if (identifier.accessPolicy.startsOn) {
+            accessPolicy.startsOn = new Date(identifier.accessPolicy.startsOn);
+          }
+        }
         res.signedIdentifiers.push({
-          accessPolicy: {
-            expiresOn: new Date(identifier.accessPolicy!.expiresOn!),
-            permissions: identifier.accessPolicy!.permissions!,
-            startsOn: new Date(identifier.accessPolicy!.startsOn!)
-          },
+          accessPolicy,
```

Comment:
What does it mean to set an ACL with only the identifier but no access policy? Is it a valid scenario?

/cc: @XiaoningLiu @jeremymeng

Reply:
It's valid since fileds in the access policy are used in combined with these in the SAS parameters. Though it makes this access policy of no use though. SDK won't add any special check for this as long as the service thinks it's OK.

Reply:
Thanks @gmantri and @ljian3377 for the details! I think it makes sense!